nntrainer/tensor/char_tensor.cpp

```diff
@@ -59,9 +59,14 @@ CharTensor::CharTensor(
   strides = dim.computeStrides();
   contiguous = true;
   initializer = Initializer::NONE;
+  qscheme = qscheme_;
 
+  NNTR_THROW_IF(scales.size() != scale_size(), std::invalid_argument)
+    << "invalid scale factor size " << scales.size();
+
+  /// @note 4 * scale_size() assumes scale factors are in full-precision fp.
   MemoryData *mem_data =
-    new MemoryData((void *)(new int8_t[dim.getDataLen()]()));
+    new MemoryData((void *)(new int8_t[dim.getDataLen() + 4 * scale_size()]()));
```

Comment:
What about updating the code like:
```suggestion
    new MemoryData((void *)(new int8_t[dim.getDataLen() + sizeof(float) * scale_size()]()));
```
If you have a plan to support different type of `scales`, it would be better to update it.

Reply:
thank you for sharing your thoughts! `sizeof(float)` makes more sense :)